API/CopilotService/__init__.py

```diff
@@ -150,7 +164,30 @@ async def _build_task_context(raw_task, token):
     return task, chunks
 
 
-def _add_context_to_queston(context, question, task_list, usage_context):
+def _find_most_relevant_command_chunk(context, question, chunks):
+    system_msg = os.environ.get("OPENAI_FIND_SIMILAR_MSG", default=DEFAULT_FIND_SIMILAR_MSG)
+    relevant_question = textwrap.dedent(
+        """\
+        text A: "{}"
+        text set B: {}
+        """.format(question, " ".join([f'"{chunk["summary"]}"' for chunk in chunks]))
+    )
+    context.custom_context.gpt_task_name = 'FIND_SIMILAR'
+    context.custom_context.estimated_question_tokens = num_tokens_from_message(question)
+    relevant_result: str = gpt_generate(context, system_msg, relevant_question, history_msg=[])
+    parts = relevant_result.rsplit('||', 1)
+    if len(parts) == 2:
+        chunk_desc, score = parts
+        if int(score) >= int(os.environ.get("DEFAULT_RELEVANT_SCORE", "8")):
+            try:
+                idx = [chunk["summary"] for chunk in chunks].index(chunk_desc)
+                return chunks[idx]
```

Comment:
If there are multiple chunks of the most similar command found, could we obtain a summary of these multiple chunks of the same command? (such as the `az vm create` command)

Reply:
The `summary` field is same when the command is split into multiple chunks, and the chunks of the same command are merged here.